tests/test_smoke.py

```diff
@@ -331,12 +331,13 @@ def test_propagate_non_default_compiler(self, capfd, basic_cmake_project):
         assert 'tools.build:compiler_executables={"c":"/usr/bin/clang","cpp":"/usr/bin/clang++"}' in out
 
     @darwin
-    def test_propagate_compiler_mac_autotools(self, capfd, basic_cmake_project):
+    @pytest.mark.parametrize("cmake_generator", ["Unix Makefiles", "Xcode"])
+    def test_propagate_compiler_mac_autotools(self, capfd, basic_cmake_project, cmake_generator):
         """Test that if the compiler is inside an XCode installation, we don't
         propagate the path if that's the compiler that would be found by default"""
         source_dir, binary_dir = basic_cmake_project
         shutil.copytree(src_dir / 'tests' / 'resources' / 'autotools_dependency', source_dir, dirs_exist_ok=True)
-        run(f"cmake -S {source_dir} -B {binary_dir} -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES={conan_provider} -DCMAKE_BUILD_TYPE=Release", check=False)
+        run(f'cmake -S {source_dir} -B {binary_dir} -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES={conan_provider} -DCMAKE_BUILD_TYPE=Release -G"{cmake_generator}"', check=False)
```

Comment:
just small comment: Xcode is multi-config generator so it takes configurations from `CMAKE_CONFIGURATION_TYPES` list instead of `CMAKE_BUILD_TYPE`